### DISCUSSION TITLE
Multi option picker for filtering data in Table.

### DISCUSSION THREAD

**Discussion by Winniepoom** [Budibase/budibase#11436] [Help] [answered]
Hello I trying to using an Multi option picker for filtering data in my table, 
I had an My SQL Employee table and had Multi Select column called Project
My Goal : is I want it to show my employee table as default and do filtering when I pick value in Project Picker .

This is an default I want it to show my employee table here. 
![image](https://github.com/Budibase/budibase/assets/107398226/819ba1d6-dfdd-4104-83a3-85b10fc47b77)

this IMG when I try pick value in Project Picker it seem to be working. 
![tempsnip](https://github.com/Budibase/budibase/assets/107398226/936aba76-26e3-4ed5-858a-e0164d367d45)

my data provider filtering
![image](https://github.com/Budibase/budibase/assets/107398226/22587938-0003-49d8-ba4b-ce1ab3a204ec)

**Comment by ConorWebb96** [comment]
Hey @Winniepoom,

Are you saying you wish for the table to be empty before using the options pickers? As for the filtering functionality this seems to be correct.

Could you possibly explain a bit more on what you are wanting here, judging from the screenshots it seems to be working as intended.

**Reply by Winniepoom** [reply]
@ConorWebb96 I wish all of my employee row  look like this before choose any option.
![tempsnip](https://github.com/Budibase/budibase/assets/107398226/a78ffab0-3116-455f-802a-8263dea69d00)

and when I pick some value it show any employee who has that value for example if I choose Project 3 my employee number 2 and 3 are filtering and show up.

**Reply by ConorWebb96** [reply]
Hey @Winniepoom,

The easiest way of doing this in my opinion would be to have an additional Data provider for your options picker to use that references your projects table. This options picker could then use this new parent project provider to generate options based on that table's contents. This could then be filtered on via the `id`. Below i'll go through demoing how to set this up. 

1. Add an options picker component with a parent data provider that links to your projects table.
<img width="100%" alt="Screenshot 2023-08-04 at 12 07 29" src="https://github.com/Budibase/budibase/assets/126772285/fcb3745d-b16a-493a-976d-480fd70b7797">
2.  Configure the options picker to use this new data source for its options. Make sure to use id as the value is unique and will make finding the linked project easily.
<img width="100%" alt="Screenshot 2023-08-04 at 12 07 36" src="https://github.com/Budibase/budibase/assets/126772285/25c966b6-fe3b-4d2d-b09b-6530e38e7137">
3. Next add the filter to your table's data provider like the screenshot below.
<img width="1462" alt="Screenshot 2023-08-04 at 12 08 10" src="https://github.com/Budibase/budibase/assets/126772285/99f788c3-096f-4ec5-9805-7f980ecdd115">

Output
![22f109d6819f48d4aa4039aeda8b3d7d](https://github.com/Budibase/budibase/assets/126772285/89729a84-7ae9-43e6-b4f5-bb24b52908c7)

I hope this helps.

**Comment by Winniepoom** [comment]
@ConorWebb96 
Thank you for your Demo,
I had an question here in your Demo you using project id for filtering.do I need to create my project id and linked to project name? 
if so I had an following question here my project column is an Multi-Select type how I'm gonna do that to link project name with project id.
Thank you 
![image](https://github.com/Budibase/budibase/assets/107398226/d3ed30b1-6d73-4c41-ae32-d3d16a9418e5)

**Reply by ConorWebb96** [reply, marked as answer]
Hey @Winniepoom,

My apologies, I assumed this field was a relationship and not a multi-select picker.

I'm now seeing the issue you are having were by default the multi-select picker when added as to the `defined filters` forces results to show on the table. I believe there is a PR out that may correct this behaviour. This can be found here; https://github.com/Budibase/budibase/pull/11262

The workaround currently for this would be to structure this in a similar way to my previous example and use relationships instead of the multi-select column type.

I believe this will fix this issue whenever it is merged into master.

I hope this helps.

**Reply by Winniepoom** [reply]
@ConorWebb96 
Thank you @ConorWebb96  I think I will wait for the fixing then.
